packages/fuels-types/src/bech32.rs

```diff
@@ -0,0 +1,102 @@
+use crate::errors::Error;
+use bech32::Variant::Bech32m;
+use bech32::{FromBase32, ToBase32};
+use core::str;
+use fuel_tx::Address;
+use std::fmt::{Display, Formatter};
+
+pub const FUEL_BECH32_HRP: &str = "fuel";
+
+#[derive(Debug, Clone)]
+pub struct Bech32Address {
+    plain_addr: Address,
```

Comment:
if `plain_addr` is pretty clear, I don't know what `encoding` corresponds to. For a reader unfamiliar with BECH32 like me, I feel like it's hard to understand what this type does "differently" than our old Address.

Reply:
One benefit of bech32 is error detection.

Given that I understood it correctly,  the encoding guarantees a distance of 5 characters between every valid address. This means you can detect a typo of up to 4 characters and report it to the user.

Checkout [BIP 0173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#rationale)